example: bungalow rental and usage

elf-pavlik opened the pull request:
### example good practices
* it includes full JSON-LD `@context` and doesn't rely on any external context
* it includes all the data under `@graph` and doesn't try to frame it as a tree with a single root #123 

### VF impact

* example proposes new action `vf:lend` to represent lend/borrow flows (instead of discussed on chat *possess*)
* for time based flows, it uses quantity to represent quantity of the involved/affected resource and uses Time to represent duration
* it uses proposed in #293 `vf:includedIn` to relate short usage during working session with longer usage between checking-in and checking-out of rented bungalow

### open issues

* we can observe start of 1 year lend event when it starts but we can't observe the whole event yet, related to #336 
* we can record fulfilled quantity of a resource for 'ongoing event' but can record the fulfilled time yet

fosterlynn commented:
>example good practices

:+1: great, we will have to start gathering useful examples for the doc

>example proposes new action vf:lend to represent lend/borrow flows

>we can observe start of 1 year lend event when it starts but we can't observe the whole event yet

I think `lend` is a positive idea for an action.  But I would like to think about it as part of a chain, not part of a hierarchy.  So there would be a vf:lend and later a vf:return or something like that.  Each would be a point in time, some rights would have been passed along and be held until another event signifies they are over.  (Like in a library, people don't usually actually return the book on the due date.  Or in this example, if they don't pay the rent, the landlord might kick them out.)  Thinking of it as a chain I think resolves the open issues you posed.  You now have some rights, and you can log usage etc.

fosterlynn commented on examples/bungalow-rental-and-usage.yaml:
```diff
@@ -0,0 +1,164 @@
+    action: lend
+    involves: tashirat:2402b57c-1c45-423a-acb2-f50961c97d65
+    commitedQuantity:
+      qudt:unit: unit:Unitless
```
I wonder if we should investigate using something besides qudt?  I heard of one from Europe I think, that might be simpler, but I'd have to track it down.  Also, I think I found something in qudt for 'each' - maybe Count or Number or something?

fosterlynn commented on examples/bungalow-rental-and-usage.yaml:
```diff
@@ -0,0 +1,164 @@
+    involves: tashirat:2402b57c-1c45-423a-acb2-f50961c97d65
+    commitedQuantity:
+      qudt:unit: unit:Unitless
+      qudt:value: 1
```
qudt is numericValue I believe

fosterlynn commented on examples/bungalow-rental-and-usage.yaml:
```diff
@@ -0,0 +1,164 @@
+    commitedQuantity:
+      qudt:unit: unit:Unitless
+      qudt:value: 1
+    commitedTime:
```
committedTime (two t's)

fosterlynn commented on examples/bungalow-rental-and-usage.yaml:
```diff
@@ -0,0 +1,164 @@
+
+  - '@id': tashirat:953b40f2-4e47-4486-afb2-402512f5269c
+    '@type': Commitment
+    under: notary:40bb78fe-28df-41a6-b698-525fa9d411d6
```
I'd like to consider something besides `under` for exchange agreements, where the commitments are a big part of the agreement, as opposed to  the agreement explaining why the commitment happened.

elf-pavlik replied:
While agreement may have some textual notes, from machine processing requirements flows that reference it with `vf:under` define it. At least as far as VF terms allow to reason.

elf-pavlik replied:
Let's deal with that as part of *agreements* milestone

fosterlynn commented on examples/bungalow-rental-and-usage.yaml:
```diff
@@ -0,0 +1,164 @@
+    observedQuantity:
+      qudt:unit: unit:Unitless
+      qudt:value: 1
+    observedTime:
```
Are you proposing changing affectedQuantity and affectedTime to observed...?  I think I might like observed better also.  But it means some changes to the handful of VF projects we have going now.  I think 2 of those will be finishing a major re-write in January, I'd like to do as much of this kind of naming cleanup as possible by say mid-January, because it will only get more painful.

fosterlynn replied:
Oops, it is already observedTime

fosterlynn commented:
>for time based flows, it uses quantity to represent quantity of the involved/affected resource and uses Time to represent duration

I think this adds unnecessary complexity to queries and processing - events with time based resources will be using a different quantity than others, also depending on the action.  I think the vf:underlyingResource keeps it simpler and cleaner.

elf-pavlik commented:
I have fixed those two typos with *commited* and *qudt:value*

> I think this adds unnecessary complexity to queries and processing - events with time based resources will be using a different quantity than others, also depending on the action. I think the vf:underlyingResource keeps it simpler and cleaner.

Could you please write a snippet with EconomicEvent *observation* of 5 hours use of 3 electric drill managed as stock with many drills in it - we have just one resource representing all those electric drills with quantity 20 or whatever and we use 3 out of them for 5 hours (no serial numbers tracked on those drills).

elf-pavlik commented:
I have updated YAML in this PR, I removed *lend* and just went with *use* everywhere. I don't know if we may need to tackle *composition* milestone first. YAML here includes 3 'levels' of nested events. In practice I think that even on days where we don't stay at the bungalow but we still have bunch of stuff here we still *use* it. In addition when we *use* it by keeping our stuff in it, but we don't stay here ourselves, someone else could stay over and *use* it as well. I wonder if we could just rely on classifications on flows to deal with those differences. On the other hand I don't think we should rely on classifications to resolve availability.

elf-pavlik commented:
> But, everything until the last `use` is really about transfers of rights, not really about use, no? You have the ability and rights to use the apartment for a year, but it is just that you can use it if you like.

I still think that making a commitment has similar effect as 'transferring very specific rights'. I think #387 might give use easier start and later we can see if we can model this one in similar way.

fosterlynn commented:
I think we should move this one to the Transfer milestone.  Although it involves composition too, but I don't thin it can be resolved without transfers.

fosterlynn commented:
Closing after v.3, please re-open if we re-look at this topic.
